analyzers/src/SonarAnalyzer.CSharp/Rules/ImplementSerializationMethodsCorrectly.cs

```diff
@@ -52,17 +53,29 @@ protected override void Initialize(SonarAnalysisContext context)
             {
                 var wrapper = (LocalFunctionStatementSyntaxWrapper)c.Node;
                 var attributes = GetSerializationAttributes(wrapper.AttributeLists, c.SemanticModel);
-
-                foreach (var attribute in attributes)
-                {
-                    c.ReportIssue(Diagnostic.Create(attributeOnLocalFunctionRule, attribute.GetLocation()));
-                }
+                ReportOnAttributes(c, attributes, "local functions");
             },
             SyntaxKindEx.LocalFunctionStatement);
 
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
```

Comment:
Coding style:  I've noticed that usually this is styled like so
```csharp
 context.RegisterSyntaxNodeActionInNonGenerated(
                c =>
                { ....

Reply:
Exception from this rule: Lambda parameter name and arrow token should be on the same line as the invocation.
```
context.RegisterSyntaxNodeAction(c =>
    {
        // Action
    }
```

I have just documented it last week based on our How we work discussion, however you are right that the coding style is not correct, the curly brackets should be indented more. I will fix it.